src/app/(dashboard)/layout.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import { Flex, Box } from '@radix-ui/themes'
+import { redirect } from 'next/navigation'
+import Sidebar from '@/components/sidebar/Sidebar'
+import Header from '@/components/header/Header'
+import { createClient } from '@/utils/supabase/server'
+
+export default async function DashboardLayout({
+  children,
+}: {
+  children: React.ReactNode
+}) {
+  const supabase = createClient()
+
+  const {
+    data: { user },
+  } = await supabase.auth.getUser()
+
+  if (!user) {
+    return redirect('/sign-in')
+  }
```

Comment:
Handling the supabase stuff in the layout file isn't something I thought about. I really like this idea.

Reply:
@davidtaing thanks mate, I'm not 100% sure if this is bullet proof way to guard the whole router but I guess it will do for now. 👍